source/robot_model/include/robot_model/Model.hpp

```diff
@@ -231,20 +248,59 @@ class Model {
 
   /**
-   * @brief Compute the inverse geometry, i.e. joint values from the pose of the end-effector
-   * @param cartesian_state containing the pose of the end-effector
-   * @return the joint state of the robot
+   * @brief parameters for the inverse geometry
+   * @param damp damping added to the diagonal of J*Jt in order to avoid the singularity
+   * @param alpha alpha €]0,1], it is used to make Newthon-Raphson method less aggressive
+   * @param gamma gamma €]0,1], represents the strength of the repulsive potential field in the Clamping Weighted Least-Norm method
+   * @param margin the distance from the joint limit on which we want to penalize the joint position
+   * @param tolerance the maximum error tolerated between the desired cartesian state and the one obtained by the returned joint positions
+   * @param max_number_of_iterations the maximum number of iterations that the algorithm do for solving the inverse geometry
    */
-  state_representation::JointPositions inverse_geometry(const state_representation::CartesianState& cartesian_state) const;
+  struct InverseGeometryParameters {
+    double damp;
+    double alpha;
+    double gamma;
+    double margin;
+    double tolerance;
+    int max_number_of_iterations;
+  };
+
+  /**
+   * @brief Compute the inverse geometry, i.e. joint values from the pose of the end-effector in a iteratively manner
+   * @param desired_cartesian_state containing the desired pose of the end-effector
+   * @param frame_name name of the frame at which we want to extract the pose
+   * @param inverse_geometry_parameters set of parameters for tuning the algorithm
+   * @return the joint positions of the robot
+   */
+  state_representation::JointPositions inverse_geometry(const state_representation::CartesianState& desired_cartesian_state,
+                                                        const std::string& frame_name = "",
+                                                        const InverseGeometryParameters& params = {
+                                                            1e-6, 0.5, 0.8, 0.07, 1e-3, 1000
+                                                        });
+
+  /**
+   * @brief Compute the inverse geometry, i.e. joint values from the pose of the end-effector
+   * @param desired_cartesian_state containing the desired pose of the end-effector
+   * @param current_joint_state current state of the robot containing the generalized position
+   * @param frame_name name of the frame at which we want to extract the pose
+   * @param inverse_geometry_parameters set of parameters for tuning the algorithm
+   * @return the joint positions of the robot
+   */
+  state_representation::JointPositions inverse_geometry(const state_representation::CartesianState& desired_cartesian_state,
+                                                        const state_representation::JointState& current_joint_state,
+                                                        std::string frame_name = "",
```

Comment:
this still need to be changed for const reference

Reply:
But it might be changed in the conditional block if the frame name is empty